src/test/java/LinkedQueueTest.java

```diff
@@ -0,0 +1,85 @@
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.EmptyStackException;
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class LinkedQueueTest {
+
+    @Test
+    @DisplayName("A new queue starts empty.")
+    void aNewQueueIsEmpty() {
+        Queue<Integer> queue = new LinkedQueue<>();
+        assertTrue(queue.isEmpty());
+    }
+
+    @Test
+    @DisplayName("An empty queue has size 0.")
+    void emptyQueueHasSizeZero() {
+        Queue<Integer> queue = new LinkedQueue<>();
+        assertEquals(0, queue.size());
+    }
+
+    @Test
+    @DisplayName("isEmpty return false when it is non empty.")
+    void nonEmptyQueueIsEmptyReturnsFalse() {
+        Queue<Integer> queue = new LinkedQueue<>();
+        queue.enqueue(99);
+        assertFalse(queue.isEmpty());
+    }
+
+    @Test
+    @DisplayName("Enqueuing items updates the size of the queue.")
+    void enqueuingUpdatesSize() {
+        Queue<Integer> queue = new LinkedQueue<>();
+        queue.enqueue(99);
+        queue.enqueue(101);
+        assertEquals(2, queue.size());
+    }
+
+    @Test
+    @DisplayName("Enqueuing one item results in it being the front.")
+    void enqueuingOneItemIsFront() {
+        Queue<Integer> queue = new LinkedQueue<>();
+        queue.enqueue(99);
+        assertEquals(99, queue.first());
+    }
+
+    @Test
+    @DisplayName("Enqueuing two items results in second not being the front.")
+    void enqueuingSecondItemIsNotFront() {
+        Queue<Integer> queue = new LinkedQueue<>();
+        queue.enqueue(99);
+        queue.enqueue(101);
+        assertNotEquals(101, queue.first());
+    }
```

Comment:
Let me know if you think this not needed. The FIFO check would do this for us.

Reply:
You may have answered your own question already. What's actually being tested here? If this fails, does the FIFO always tests fail? If the FIFO test fails, will this always fail?